src/main/java/com/zaxxer/hikari/proxy/ConnectionProxy.java

```diff
@@ -147,8 +147,9 @@ public final void markCommitStateDirty()
       return statement;
    }
 
-   private final void closeOpenStatements()
+   private final boolean closeOpenStatements()
    {
+      boolean success = true;
```

Comment:
can't. it has to return true in all cases except in case of exception that sets evict in 'any' statement closing